src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/IngestSdkProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal.ingestsdk;
+
+/**
+ * This is a singleton provider that provides global access to the static managers of the ingest sdk
+ */
+public class IngestSdkProvider {
+  // manages the streaming ingest sdk client
+  public static StreamingClientManager streamingClientManager = new StreamingClientManager();
```

Comment:
may be this is just me but this public variable makes it vulnerable to change. I would may be wrap this to a public getter and make this private. public getter can also ensure this is not null before returning.

Reply:
making it vulnerable to change was part of the design - we need a way to set it for testing because its a singleton. However it is a little too vulnerable, especially since that vulnerability is not documented. I'll make the public getter but that requires a constructor that breaks the singleton. 

out of scope for this pr, but adding a global isTest boolean that is false by default but is flipped during testing could help enforce this into a 'singleton' in production